src/apps/seo/src/app/components/RedirectsDialogProvider/ChangeDialog.tsx

```diff
@@ -0,0 +1,183 @@
+import { FC } from "react";
+import LoadingButton from "@mui/lab/LoadingButton";
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogTitle,
+  Typography,
+  Box,
+} from "@mui/material";
+
+import { DialogContent } from "@mui/material";
+import { ShuffleVariant } from "@zesty-io/material";
+import ArrowDownwardIcon from "@mui/icons-material/ArrowDownward";
+import {
+  RedirectsCodes,
+  RedirectsTargetType,
+} from "../../../../../../shell/services/types";
+import { useRedirectsDialog } from ".";
+import { useDispatch } from "react-redux";
+import { notify } from "../../../../../../shell/store/notifications";
+
+export type ChangeRedirectProps = {
+  targetType: RedirectsTargetType;
+  target: string;
+  path: string;
+  code: RedirectsCodes;
+};
+
+export type ChangeDialogProps = {
+  open: boolean;
+  onClose: () => void;
+  redirect?: ChangeRedirectProps | null;
+  newPath?: string;
+};
+
+export const ChangeDialog: FC<ChangeDialogProps> = ({
```

Comment:
I suggest naming this component into something more descriptive of what it does, like in this case, what kind of change is this dialog doing or is for. ChangeDialog to me kind of sounds unclear on what it change the dialog does.